app/controllers/file.js

```diff
@@ -67,17 +67,18 @@ FileController.prototype.downloadFile = function(opts) {
           this.hasDownloadError = false;
           if (res.statusCode == 200) {
             let fileSize = res.headers["content-length"];
-            fileSystem.isThereAvailableSpace(this.logger, (isThereAvailableSpace) => {
-              if(isThereAvailableSpace) {
-
-                fileSystem.addToDownloadTotalSize(fileSize);
-                this.writeFile(res);
-                this.emit("downloading");
-
-              } else {
-                this.emit("insufficient-disk-space", fileSize);
-              }
-            }, spaceInDisk, fileSize);
+            if (!spaceInDisk) {
```

Comment:
I don't think this should proceed if spaceInDisk === 0 right?

Reply:
I agree, it should distinguish between 0 ( no space ) and false ( error occurred )